app/controllers/wbs_controller.rb

```diff
@@ -13,10 +14,20 @@ def index
         render :layout => !request.xhr?
       }
       format.api  {
-        @issues = @project.issues.visible
-                    .where
-                      .not(tracker_id: RedmineWbs.excluded_tracker_ids.reject { |c| c.empty? })
-                    .order("#{Issue.table_name}.root_id ASC, #{Issue.table_name}.lft ASC")
+        query = "
+          SELECT I.*
+            FROM issues I
+           WHERE I.`project_id` = ?
+             AND I.`root_id` NOT IN (
+              SELECT I.`id`
+                FROM `issues` I
+                JOIN `trackers` T ON t.`id` = I.`tracker_id`
+               WHERE I.`root_id` = I.`id` AND I.`tracker_id` IN (?) OR T.`fields_bits` & 64
+            )
+          ORDER BY I.`root_id` ASC, I.`lft` ASC
+        "
+
+        @issues = Issue.find_by_sql([query, @project.id, @excluded_tracker_ids])
```

Comment:
It think you could avoid the method `build_expluded_tracker_ids` if you create this query with Rails' query builder. See https://stackoverflow.com/questions/5483407/subqueries-in-activerecord for an example.